Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3553,6 +3553,16 @@ void RenderBox::computePreferredLogicalWidths(const Length& minWidth, const Leng
 {
     if (shouldComputeLogicalHeightFromAspectRatio()) {
         auto [logicalMinWidth, logicalMaxWidth] = computeMinMaxLogicalWidthFromAspectRatio();
+        logicalMinWidth = std::max(logicalMinWidth - borderAndPadding, LayoutUnit(0));
+        logicalMaxWidth = std::max(logicalMaxWidth - borderAndPadding, LayoutUnit(0));
+
+        if (style().width().isFixed()) {
```

Comment:
Use `style().logicalWidth()` instead of `style().width()` here? And if `style().logicalWidth().isFixed()`, there is no need to calculate transferred min/max width, so we could move it to the `if` constraint.
We have `RenderBox::constrainLogicalMinMaxSizesByAspectRatio` to handle different scenarios of transferred min/max size. Maybe in the follow-up patch, we can try to reuse this function here?

Reply:
Changed! And yes that makes sense. There are still some flex box WPT tests that are failing so I can try to add that along with one of those patches.